src/sirhelpers.c

```diff
@@ -314,25 +314,28 @@ int _sir_fopen(FILE* restrict* restrict streamptr, const char* restrict filename
 
 struct tm* _sir_localtime(const time_t* restrict timer, struct tm* restrict buf) {
     if (_sir_validptr(timer) && _sir_validptr(buf)) {
-#if defined(__HAVE_STDC_SECURE_OR_EXT1__)
+#if defined(__HAVE_STDC_SECURE_OR_EXT1__) && !defined(__EMBARCADEROC__)
 # if defined(__WIN__)
-        errno_t ret = (errno_t)localtime_s(buf, timer);
+        errno_t ret = localtime_s(buf, timer);
         if (0 != ret) {
             (void)_sir_handleerr(ret);
             return NULL;
         }
-
-        return buf;
-# else /* __WIN__ */
+# else
         struct tm* ret = localtime_s(timer, buf);
-        if (!ret)
-            (void)_sir_handleerr(errno);
-
-        return ret;
+        if (!ret) {
+            (void)_sir_handleerr(ret);
```

Comment:
I don't think you meant to do this–`ret` is `struct tm*`. My man page says:

```
When encountering an error, these
     functions return NULL and set errno to an appropriate value
```

So it was right before...

Reply:
Looking at https://en.cppreference.com/w/c/chrono/localtime for the localtime_s function, it says it returns "a copy of the buf pointer, or null pointer on error (which may be a runtime constraint violation or a failure to convert the specified time to local calendar time)".  So checking for `!ret` would be correct, no? I should be using errno though in the error reporting here, fixed.

Reply:
Kind of a pain, I don't have any EXT1 UNIXen here to actually test that path... I think maybe VxWorks?  Grr.